catalyst-gateway/bin/src/service/api/cardano/cip36/filter.rs

```diff
@@ -444,29 +473,60 @@ pub async fn get_all_registrations(
             continue;
         };
 
+        let stake_pub_key = match Ed25519HexEncodedPublicKey::try_from(row.stake_public_key.clone())
+        {
+            Ok(stake_pub_key) => Some(stake_pub_key),
+            Err(err) => {
+                error!("Corrupt valid registration {:?}", err);
+                // This SHOULD not happen, valid registrations should be infallible.
+                // If it happens, there is an indexing issue.
+                continue;
+            },
+        };
+
+        let payment_address = match Cip19ShelleyAddress::try_from(row.payment_address) {
+            Ok(payment_addr) => Some(payment_addr),
+            Err(err) => {
+                error!(
+                    "Corrupt valid registration {:?}\n Stake pub key:{:?}",
+                    err, stake_pub_key
+                );
+                continue;
+            },
+        };
+
+        let vote_pub_key = match Ed25519HexEncodedPublicKey::try_from(row.vote_key) {
+            Ok(vote_pub_key) => Some(vote_pub_key),
+            Err(err) => {
+                error!(
+                    "Corrupt valid registration {:?}\n Stake pub key:{:?}",
+                    err, stake_pub_key
+                );
+                continue;
+            },
+        };
+
         let cip36 = Cip36Details {
             slot_no: SlotNo::try_from(slot_no)?,
```

Comment:
this should be infallible or its an error

Reply:
resolved.